xbmc/filesystem/ZipManager.cpp

```diff
@@ -100,6 +100,16 @@ bool CZipManager::GetZipList(const CURL& url, std::vector<SZipEntry>& items)
   // Don't need to look in the last 18 bytes (ECDREC_SIZE-4)
   // But as we need to do overlapping between blocks (3 bytes),
   // we start the search at ECDREC_SIZE-1 from the end of file
+  if (fileSize == 0)
+  {
+    CLog::Log(LOGERROR, "ZipManager: Couldn't get zip file length");
+    return false;
```

Comment:
What about (invalid) zip files which have 0 filesize? In that case the error message is not correct.